src/components/Cluster/ClusterDetail/UserInstalledApps/InstalledApp.tsx

```diff
@@ -34,61 +34,48 @@ const App = styled.div`
   }
 `;
 
-interface IAppMetadata {
-  name: string;
-}
-
-interface IAppSpec {
-  version?: string;
-}
-
-interface IApp {
-  logoUrl?: string;
-  metadata: IAppMetadata;
-  spec?: IAppSpec;
-}
-
 interface IInstalledAppProps {
-  app: IApp;
-  iconErrors?: object;
-  onIconError: () => void;
+  name: string;
+  version: string;
+  onIconError: React.ReactEventHandler<HTMLImageElement>;
   onClick: (e: React.MouseEvent) => void;
+  logoUrl?: string;
+  iconErrors?: Record<string, boolean>;
```

Comment:
Made the interface a bit more flexible, so I could reuse it in the MAPI components